src/components/RoomView/RoomView.jsx

```diff
@@ -28,42 +28,50 @@ const RoomView = () => {
 
   // frontend state
   const roomId = useSelector((state) => state.player.roomId);
+  const playerId = useSelector((state) => state.player.playerId);
   const username = useSelector((state) => state.player.username);
   const allPlayers = useSelector((state) => state.allPlayers.allPlayers);
   const [loading, setLoading] = useState(false);
 
-  // firebase room  & players reference
-  let roomRef = ref(database, "rooms/" + roomId);
-  const allPlayersRef = ref(database, "players/");
+ // firebase room  & players reference
+ let roomRef = ref(database, "rooms/" + roomId);
+ let playersInRoomRef = ref(database, "rooms/" + roomId + '/players/');
 
   useEffect(() => {
+    console.log('in room view use effect')
     // on loading page if no room or name, send back to join page
     if (roomId === "" || username === "") {
       navigate("/");
     } else {
       console.log("joined room!");
     }
 
-    // whenever users are added
-    onValue(allPlayersRef, (snapshot) => {
-      setLoading(true);
-      const data = snapshot.val();
+    //when a user joins room, this checks to see if it exists 
+    get(roomRef).then((snapshot) => {
+      const doesRoomExist = snapshot.exists()
+        if (doesRoomExist) {
+          console.log("room already created, just add the player!");
+          // playerId is key in the room/roomId/players/playerId, so we creating new player obj
+          set(child(playersInRoomRef, playerId), { playerId, username })
+        } else {
+          // create the room, with players and host
+          console.log('room not created yet. make one!')
+          set(roomRef, {roomId: roomId, host: {playerId, username}, players: { [playerId]: {playerId, username }}})
```

Comment:
one of the ways i tried moving players into rooms prior to coming to the current refactor was using push. In the docs, they use 'push' to add a post to a blog which is what i used as a template. firebase gave me trouble when trying to disconnect the user so i abandoned that approach. i think it might be possible but i just havent found the answer yet either.

re: the key/value  also having an id, we can take that out i just thought it might be helpful at some point